src/utils/getComponentName.js

```diff
@@ -1,6 +1,6 @@
 // @flow
 import type { ComponentType } from 'react';
 
-export default function getComponentName(target: ComponentType<*>): string {
-  return target.displayName || target.name || 'Component';
+export default function getComponentName(target: ComponentType<*> | string): string {
+  return typeof target === "string" ? target : target.displayName || target.name || 'Component';
```

Comment:
this can be reverted as well

Reply:
I don't think so, without it the generated display name will be ```Styled(Component)``` with mocked components because string doesn't have either displayName || name

Reply:
It won't be because you're accounting for the uppercase string name in `isTag`. It'll end up as `styled.YourMockedName`

Reply:
No, it won't, because ```isTag``` will return false for such components (check for first lower case letter will fail)